applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -622,39 +574,32 @@ void SmallStrainUPwDiffOrderElement::
     const SizeType NumUNodes = rGeom.PointsNumber();
     const SizeType NumPNodes = mpPressureGeometry->PointsNumber();
 
-    // const bool IgnoreUndrained = GetProperties()[IGNORE_UNDRAINED];
-    // const SizeType ElementSize = (IgnoreUndrained ? NumUNodes * Dim : NumUNodes * Dim + NumPNodes);
     const SizeType ElementSize = NumUNodes * Dim + NumPNodes;
 
     if ( rResult.size() != ElementSize )
         rResult.resize( ElementSize, false );
 
     SizeType Index = 0;
 
-    if (Dim > 2) {
-        for ( SizeType i = 0; i < NumUNodes; ++i ) {
-            rResult[Index++] = rGeom[i].GetDof( DISPLACEMENT_X ).EquationId();
-            rResult[Index++] = rGeom[i].GetDof( DISPLACEMENT_Y ).EquationId();
-            rResult[Index++] = rGeom[i].GetDof( DISPLACEMENT_Z ).EquationId();
-        }
-    } else {
-        for ( SizeType i = 0; i < NumUNodes; ++i ) {
-            rResult[Index++] = rGeom[i].GetDof( DISPLACEMENT_X ).EquationId();
-            rResult[Index++] = rGeom[i].GetDof( DISPLACEMENT_Y ).EquationId();
+    for ( SizeType i = 0; i < NumUNodes; ++i ) {
+        rResult[Index] = rGeom[i].GetDof( DISPLACEMENT_X ).EquationId();
+        ++Index;
+        rResult[Index] = rGeom[i].GetDof( DISPLACEMENT_Y ).EquationId();
+        ++Index;
+        if (Dim > 2) {
+            rResult[Index] = rGeom[i].GetDof( DISPLACEMENT_Z ).EquationId();
+            ++Index;
         }
     }
+    for ( SizeType i = 0; i < NumPNodes; ++i ){
+        rResult[Index] = rGeom[i].GetDof( WATER_PRESSURE ).EquationId();
+        ++Index;
+    }
```

Comment:
I like the improvement in this part, let's keep in mind for a next PR that we can extract a function here, since it seems identical for velocity and acceleration